library/src/main/java/com/smartnsoft/webviewfragment/WebViewFragment.java

```diff
@@ -343,6 +305,40 @@ public void onClick(View view)
     }
   }
 
+  protected void onRetrieveBusinessObjects()
+  {
+    url = getArguments().getString(WebViewFragment.PAGE_URL_EXTRA);
+    defaultErrorMessage = getArguments().getString(WebViewFragment.DEFAULT_ERROR_MESSAGE_EXTRA);
+
+    if (getArguments().getSerializable(WebViewFragment.ERROR_MESSAGES_TABLE_EXTRA) instanceof Map)
+    {
+      errorMessagesTable = (Map<Integer, String>) getArguments().getSerializable(WebViewFragment.ERROR_MESSAGES_TABLE_EXTRA);
+    }
+  }
+
+  protected void onFulfillDisplayObjects()
+  {
+    if (getArguments().getString(WebViewFragment.SCREEN_TITLE_EXTRA, null) == null)
+    {
+      try
+      {
+        if (getActivity() instanceof AppCompatActivity)
+        {
+          ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(webView.getTitle());
+        }
+      }
+      catch (Exception exception)
+      {
+        warn("Cannot set the title", exception);
+      }
+    }
+
+    configureWebView();
+    refreshMenu();
+  }
+
+  protected abstract void warn(String message, Throwable throwable);
```

Comment:
@ludovicroland Why is this method abstract ?

Reply:
Hi @n0tnull,

This method is abstract in order to allow developers to use the library they want in order to log the warning (logentries, log4j, Timber and so on).

Reply:
Ok